File: ProjJava/app/src/main/java/com/example/projjava/MyReceiver.java
package com.example.projjava;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Debug;
import android.util.Log;

import java.io.IOException;

public class MyReceiver extends BroadcastReceiver {
    private final static String TAG = "broadcastR";
    @Override
    public void onReceive(Context context, Intent intent) {
        Bencher.getInstance().runGC();
        Bencher.getInstance().logStart(TAG);
        Bencher.getInstance().dumpHeap("/sdcard/prev.hprof");

        String operation = intent.getStringExtra("function");
        Intent intent2 = new Intent(context, MyService.class);
        intent2.putExtra("function", operation);
        Log.i(TAG, "Launching service (API VERSION: "+ Build.VERSION.SDK_INT+")");
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(intent2);
        } else {
            context.startService(intent2);
        }
        Log.i(TAG, "Received broadcast for "+operation+" started");
    }
}

File: projKotlin/app/src/main/java/com/example/projkotlin/MyReceiver.kt
package com.example.projkotlin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class MyReceiver : BroadcastReceiver() {

    val TAG :String = "BroadcastReceiver"
    override fun onReceive(context: Context, intent: Intent) {
        // This method is called when the BroadcastReceiver is receiving an Intent broadcast.
        BencherHelper.runGC()
        BencherHelper.dumpHeap("/sdcard/prevkt.hprof")
        BencherHelper.logStart(TAG)

        val operation = intent.getStringExtra("function")
        val intent2 = Intent(context, MyService::class.java)
        intent2.putExtra("function", operation)
        Log.i(TAG, "waiting for 5 seconds for unplugging the phone")
        try {
            Thread.sleep(5000)
        } catch (e: Exception) {
            Log.i(TAG, "sleep time interrupted")
        }
        Log.i(TAG, "Launching service")
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(intent2)
        } else {
            context.startService(intent2)
        }
        Log.i(TAG, "Received broadcast for $operation started")
    }
}
